remix-tests/src/compiler.ts

```diff
@@ -0,0 +1,126 @@
+import fs from './fileSystem'
+import async from 'async'
+var path = require('path')
+let RemixCompiler = require('remix-solidity').Compiler
+import { SrcIfc } from './types'
+
+function regexIndexOf (inputString: string, regex: RegExp, startpos: number = 0) {
+    var indexOf = inputString.substring(startpos).search(regex)
+    return (indexOf >= 0) ? (indexOf + (startpos)) : indexOf
+}
+
+function writeTestAccountsContract (accounts: string[]) {
+    var testAccountContract = require('../sol/tests_accounts.sol.js')
+    var body = 'address[' + accounts.length + '] memory accounts;'
+    if (!accounts.length) body += ';'
+    else {
+        accounts.map((address, index) => {
+            body += `\naccounts[${index}] = ${address};\n`
+        })
+    }
+    return testAccountContract.replace('>accounts<', body)
+}
+
+var userAgent = (typeof (navigator) !== 'undefined') && navigator.userAgent ? navigator.userAgent.toLowerCase() : '-'
+var isBrowser = !(typeof (window) === 'undefined' || userAgent.indexOf(' electron/') > -1)
+
+// TODO: replace this with remix's own compiler code
+export function compileFileOrFiles(filename: string, isDirectory: boolean, opts: any, cb: Function) {
```

Comment:
Maybe use `Promise` and `async function` instead of callback if possible.

Reply:
I think we should use `async` `functions` for all our new Typescript versions of `modules`. But those will go in different PR.

Reply:
No, don't use `async` if there is no asynchrone method inside. Expect if you really want to match the `Promise` interface for all your methods for some external reasons.